src/artm_tests/topic_seg_test.cc

```diff
@@ -0,0 +1,123 @@
+#include <memory>
+
+#include "gtest/gtest.h"
+
+#include "artm/cpp_interface.h"
+#include "artm/core/common.h"
+#include "artm/core/instance.h"
+
+#include "artm_tests/test_mother.h"
+#include "artm_tests/api.h"
+
+TEST(Regularizer, TopicSegmentationPtdw) {
+    int n_topics = 5;
+    int n_background_topics = 2;
+    ::artm::MasterModelConfig master_config = ::artm::test::TestMother::GenerateMasterModelConfig(n_topics);
+    master_config.set_cache_theta(true);
+    master_config.set_num_document_passes(0);
+    
+    artm::MasterModel master_model_1(master_config);
+    artm::MasterModel master_model_2(master_config);
+    ::artm::test::Api api_1(master_model_1);
+    ::artm::test::Api api_2(master_model_2);
+    
+    ::artm::Batch batch;
+    batch.set_id(artm::test::Helpers::getUniqueString());
+    batch.add_token("aaaa0");  //0
+    batch.add_token("bbbb1");  //1
+    batch.add_token("cccc2");  //2
+    batch.add_token("dddd3");  //3
+    batch.add_token("eeee4");  //4
+    batch.add_token("ffff5");  //5
+    artm::Item* item = batch.add_item();
+    item->set_id(0);
```

Comment:
std::vector temp = { 1, 2, 3, 4, 5 ...}
for (auto e : temp) {
  item->add_token_id(e);
  item->add_token_weight(1.0);
}

smth like this